orchestra/cmds/update.py

```diff
@@ -85,31 +86,54 @@ def handle_update(args):
         """)
         logger.error(failed_git_pull_suggestion)
 
+    if failed_clones:
+        formatted_failed_clones = "\n".join([f"  {repo}" for repo in failed_clones])
+        failed_git_clone_suggestion = dedent(f"""
+                Could not clone the following repositories:
+                {formatted_failed_clones}
 
-def pull_binary_archive(name, config):
-    binary_archive_path = os.path.join(config.binary_archives_dir, name)
-    logger.info(f"Pulling binary archive {name}")
-    result = git_pull(binary_archive_path)
-    return result
+                Suggestions:
+                    - check your network connection
+                    - check your ssh and git configuration (try manually cloning the repositories)
+                """)
+        logger.error(failed_git_clone_suggestion)
+
+    if failed_pulls or failed_clones:
+        return 1
 
 
 def clone_binary_archive(name, url, config):
-    logger.info(f"Trying to clone binary archive from remote {name} ({url})")
+    """Clones a binary archive. Returns a boolean value representing the operation's success."""
     binary_archive_path = os.path.join(config.binary_archives_dir, name)
-    env = dict(os.environ)
+    env = os.environ.copy()
     env["GIT_SSH_COMMAND"] = "ssh -oControlPath=~/.ssh/ssh-mux-%r@%h:%p -oControlMaster=auto -o ControlPersist=10"
     env["GIT_LFS_SKIP_SMUDGE"] = "1"
-    result = subprocess.run(["git", "clone", url, binary_archive_path], env=env)
-    if result.returncode:
-        logger.info(f"Could not clone binary archive from remote {name}!")
+    try:
+        quiet_run_internal_subprocess(
+            ["git", "clone", url, binary_archive_path],
+            environment=env,
+        )
+        return True
+    except OrchestraException:
+        return False
```

Comment:
I'm not sure how this is better than:
```python
return try_get_process_output(...) is not None
```
I thought I persuaded yesterday with the non-local handling of errors thing.